src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -275,12 +240,78 @@ private RestoreData CreateRestoreData(ProjectRestoreInfo restoreInfo, bool succe
                 succeeded: succeeded && lastWriteTime != DateTime.MinValue);
         }
 
+        protected virtual bool IsProjectConfigurationVersionOutOfDate(IReadOnlyCollection<PackageRestoreConfiguredInput> configuredInputs)
+        {
+            Assumes.Present(_project.Services.ActiveConfiguredProjectProvider);
+            Assumes.Present(_project.Services.ActiveConfiguredProjectProvider.ActiveConfiguredProject);
+
+            if (configuredInputs is null)
+            {
+                return false;
+            }
+
+            ConfiguredProject activeConfiguredProject = _project.Services.ActiveConfiguredProjectProvider.ActiveConfiguredProject;
+
+            IComparable? activeProjectConfigurationVersionFromConfiguredInputs = null;
+            foreach (PackageRestoreConfiguredInput configuredInput in configuredInputs)
+            {
+                if (configuredInput.ProjectConfiguration.Equals(activeConfiguredProject.ProjectConfiguration))
+                {
+                    activeProjectConfigurationVersionFromConfiguredInputs = configuredInput.ConfiguredProjectVersion;
+                }
+            }
+            
+            if (activeProjectConfigurationVersionFromConfiguredInputs is null ||
+                activeConfiguredProject.ProjectVersion.IsLaterThan(
+                    activeProjectConfigurationVersionFromConfiguredInputs))
+            {
+                return true;
+            }
+
+            if (configuredInputs.Count == 1)
+            {
+                return false;
+            }
+
+            foreach (ConfiguredProject loadedProject in activeConfiguredProject.UnconfiguredProject.LoadedConfiguredProjects)
+            {
+                foreach (PackageRestoreConfiguredInput configuredInput in configuredInputs)
+                {
+                    if (loadedProject.ProjectConfiguration.Equals(configuredInput.ProjectConfiguration) &&
+                        loadedProject.ProjectVersion.IsLaterThan(configuredInput.ConfiguredProjectVersion))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
         public Task LoadAsync()
         {
             _enabled = true;
 
             EnsureInitialized();
 
+            if (!_wasSourceBlockContinuationSet)
+            {
+                _wasSourceBlockContinuationSet = true;
+                
+                // Inform the NuGet restore service when there will be no further updates so it can cancel any pending work.
+                _ = SourceBlock.Completion.ContinueWith(t =>
```

Comment:
Q: Is this a pattern now to assign a discard for methods that return a value instead of just calling them like a `void` method?

Reply:
It's a language feature, actually. You can assign to "_" without having to declare, and it signals that a value is being returned but you've made a conscious choice to ignore it.